pkg/tfbridge/provider.go

```diff
@@ -1013,15 +1009,20 @@ func (p *Provider) Create(ctx context.Context, req *pulumirpc.CreateRequest) (*p
 	label := fmt.Sprintf("%s.Create(%s/%s)", p.label(), urn, res.TFName)
 	glog.V(9).Infof("%s executing", label)
 
+	props, err := plugin.UnmarshalProperties(req.GetProperties(),
+		plugin.MarshalOptions{Label: label, KeepUnknowns: true, SkipNulls: true})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshaling %s's new property state", urn)
+	}
 	// To get Terraform to create a new resource, the ID must be blank and existing state must be empty (since the
 	// resource does not exist yet), and the diff object should have no old state and all of the new state.
-	config, assets, err := UnmarshalTerraformConfig(ctx,
-		p, req.GetProperties(), res.TF.Schema(), res.Schema.Fields,
-		fmt.Sprintf("%s.news", label))
+	inputs, assets, err := makeTerraformInputsNoDefaultsWithMaxItemsOneDefaults(
+		ctx, nil, nil, nil, props, res.TF.Schema(), res.Schema.Fields,
+	)
 	if err != nil {
-		return nil, errors.Wrapf(err, "preparing %s's new property state", urn)
+		return nil, errors.Wrapf(err, "preparing %s's new property inputs", urn)
 	}
-
+	config := MakeTerraformConfigFromInputs(ctx, p.tf, inputs)
```

Comment:
I've inlined UnmarshalTerraformConfig here since this was the only place it was used and it limited customization here. `makeTerraformInputsNoDefaultsWithMaxItemsOneDefaults` is a new `makeTerraformInputs` variant used only here. This is meant to be the only logic change to this method.